stdlib/public/stubs/Stubs.cpp

```diff
@@ -323,6 +323,18 @@ swift::swift_stdlib_readLine_stdin(unsigned char **LinePtr) {
 #endif
 }
 
+#if __has_attribute(__mode__)
+#define SWIFT_MODE_DI __attribute__((__mode__(DI)))
+#define SWIFT_MODE_TI __attribute__((__mode__(TI)))
+#else
+#define SWIFT_MODE_DI
+#define SWIFT_MODE_TI
```

Comment:
Please make this a hard error.  Without this, the types are completely the wrong size and you do not get any type of warning/error that things went wrong.

Reply:
The problem with a hard error is that I moved this out of the `#if` blocks, since the definitions are used by multiple disjoint `if` blocks. If we hard error, then that applies to *every* platform, even ones that will never need to use the types.

The alternatives would be to either warn or to duplicate these definitions and move them back into each `#if` block. The problem that then arises is that the definitions become conditional on what `#if` blocks we'd already encountered, unless we `#define`d and `#undef`d the types within each block.

Reply:
I've addressed this by adding a 3-line check for `__mode__` support within each `#if` block.